waspc/data/Generator/templates/react-app/src/actions/index.js

```diff
@@ -27,6 +29,18 @@ function makeOptimisticUpdatesConfig(optimisticUpdatesConfig) {
   }))
 }
 
+function makeOptimisticUpdateMutationFn(actionFn, optimisticUpdatesConfig) {
+  return function optimisticallyUpdateQueries(item) {
+    const optimisticUpdateTuples = optimisticUpdatesConfig.map(
+      ({ getQuery, updateQuery }) => ({
+        queryKey: getQuery(item), 
+        updateQueryFn: (old) => updateQuery(item, old),
+      })
+    ) 
+    return actionFn.internal(item, optimisticUpdateTuples)
```

Comment:
Ok, so instead of calling an action, we call its internal thing, while passing info about optimistic updates, so that action becomes "smarter"?

I find this whole code really hard to read! The fact there is no types makes it super hard. Let's bomb it with comments describing types, utilizing `jsdoc` and its `@param`, `@typedef`, `@type` and similar, until we switch to TS?